documentation (#51)
Rappsilber-Laboratory/xiFDR: README.md

### Loading crosslinking MS search results

The top half of the input tab relates to reading in results, while the bottom relates to mapping column names, which is needed only in case of loading results that are not coming from xiSEARCH.

#### Loading search results from xiSEARCH

To load results from xiSEARCH, only use the top half of the "input" tab containing csv-file, xi config and fasta.

xiSEARCH outputs results in a .csv file that contains all crosslinked peptide spectrum matches (CSMs) and scores. It includes target-target matches, target-decoy and decoy-decoy matches.

> Select the xiSEARCH .csv results file. The .fasta database and the .config file used in the search should also be uploaded in the respective boxes if available.

Reviewer: ```suggestion
Select the xiSEARCH .csv results file. The .fasta database and the .config file used in the search should also be uploaded in the respective boxes if available. Fasta and config file are required  to write out results in .mzIdentML format. **These should be set before pressing "read".**
```

Any prefilters on spectral quality (see below) should be set by ticking the "filter" box.

> At this stage, one can select .fasta file and .config files from xiSEARCH, which are required to write out results in .mzIdentML format. **These should be set before pressing "read".**
> 

Reviewer: ```suggestion
```

#### prefilters
xiSEARCH provides many features of CSMs that may be used to prefilter the results prior to FDR estimation. Doing so equally on targets and decoys prior to FDR estimation retains the accuracy of FDR, while doing score filtering or other filters post FDR estimation generates results with unknown error rates. We recommend doing this only after having a look at the FDR-filtered results without any prefilters. 

> If running without prefilters already generates warnings about not enough TTs at the level of interest (links or protein pair) when running FDR calculations without any prefilters, prefilters should NOT be used.

Reviewer: I don't think this should be quite as strict - there are probably cases where pre-filter actually filter out false positives selectively enough that a previously unreliable FDR becomes reliable.
```suggestion
If running without pre-filters already generates warnings about not enough TTs at the level of interest (links or protein pair), pre-filters are often not making the FDR more reliable.
```

In the "FDR settings", one can perform the actual FDR filtering. 

By default, the view is set to "reduced FDR", which shows just the basic settings. The cutoff is set at 5% at the residue pair level, meaning the  error will be propagated so that 5% of the residue pairs correspond to a wrong/random match. The "boosting" features is enabled (see below for more details). These are perfectly acceptable FDR filtering settings for experiments aimed at characterising the crosslinks in a purified protein complex and should give a good idea of the number of crosslinks detectable with reasonable certainty in the sample. In analyses of cellular fractions or searches with hundreds of proteins in the database, it is advisable to also include an FDR cutoff at the "Protein Pairs" level. Similarly, in analyses devoted to method development on the quality of spectra, a filter at the "PSM" is advised.

> Whenever multiple cutoffs are selected (e.g. both residue pairs and PPI level at 5%), xiSEARCH will report only the entries at lower levels that also pass the upper level. 
> For example, only the residue pairs that will pass both the residue pair and PPI level thresholds will be considered as passing the FDR.

Reviewer: Maybe a adding bit of a warning:
```suggestion
Whenever multiple cutoffs are selected (e.g. both residue pairs and PPI level at 5%), xiSEARCH will report only the entries at lower levels that also pass the upper level. 
For example, only the residue pairs that will pass both the residue pair and PPI level thresholds will be considered as passing the FDR. Note that the in result there will be fewer decoy residue pairs then the residue pair indicates however one can not use the still present decoys as an estimates of remaining false positive positive residue pairs (at this point decoys FDR would underestimate the FPR). 
```

Author: Sorry the english is not clear here.

I think you should revise to

Notice that in these cases the results files at lower levels will have less decoy lines than the stated FDR threshold at that particular level (as only the targets and decoys passing also the higher level are reported). Thus, these files cannot be used to recompute an estimate of the false positive rate, as the decoy-based FDR would drastically underestimate the FPR. Thus: with both residue pair and PPI level FDR, the number of decoys in the "Links" csv cannot be used as a basis of analyses that recompute apparent decoy-based FDR. For these analyses, the use of a threshold at a single level (the one of interest) is recommended.

Author: edited PR now
